client/src/components/StockDetailPage/BuySellMenu.jsx

```diff
@@ -17,30 +17,37 @@ const BuySellMenu = (props) => {
   }
   numStock = numStock || 0;
   const displaySell = numStock > 0 ? true : false;
-  const history = useHistory();
-  const handleClick = () => {
-    history.push('/trade');
+  const handleClick = (event) => {
+    const action = event.target.value;
+    props.updateTradeAction(action);
   };
   return (
     <div className='BuySellMenu'>
       {displayBuy ?
-        <button
-          className='buyButton'
-          onClick={handleClick}
-        >
-            Buy
-        </button> :
-        null}
+        <Link to='/trade'>
+          <button
+            className='buyButton'
+            onClick={handleClick}
```

Comment:
I learned something new.  I have always invoked those functions using onClick={(event) => handleClick(event}.  Your way is better. Thanks!